### PR TITLE
Adding support for true and apparent species lists

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
Part of building up the infrastructure required for modeling electrolyte systems, this PR adds the code necessary to switch the basic of the component list between true and apparent species. Also included is updating some other modules to make use of the new phase and component list properties to facilitate this switching, plus some basic tests.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+            "HCO3-", "K+", "H2O", "CO2", "KHCO3", "N2"]
+
+        assert m.fs.props.true_phase_component_set == [
+            ("Liq", "HCO3-"), ("Liq", "K+"), ("Liq", "H2O"), ("Liq", "CO2"),
+            ("Vap", "H2O"), ("Vap", "CO2"), ("Vap", "N2")]
+        assert m.fs.props.apparent_phase_component_set == [
+            ("Liq", "H2O"), ("Liq", "CO2"), ("Liq", "KHCO3"),
+            ("Vap", "H2O"), ("Vap", "CO2"), ("Vap", "KHCO3"), ("Vap", "N2")]
+
+        assert m.fs.state[1].component_list is m.fs.props.true_species_set
```
Comment (TimBartholomew) [inline idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py:380]: This assertion confuses me, why is the component list and true species set the same for the state block? The state block will still include the apparent species, right?

File: idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+            assert p in ["Liq", "Vap"]
+        assert isinstance(m.fs.state[1].phase_frac, Var)
+        assert len(m.fs.state[1].phase_frac) == 2
+        for p in m.fs.state[1].phase_frac:
+            assert p in ["Liq", "Vap"]
+
+        assert isinstance(m.fs.state[1].mole_frac_comp, Var)
+        assert len(m.fs.state[1].mole_frac_comp) == 5
+        for j in m.fs.state[1].mole_frac_comp:
+            assert j in ["HCO3-", "K+", "H2O", "CO2", "N2"]
```
Comment (TimBartholomew) [inline idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py:404]: Ok, so the state block does not explicitly include apparent species, right? How will users access the apparent species?

**Review by TimBartholomew — COMMENTED** [review]
Looks good. I have an implementation question for apparent species on the state block as follows.

File: idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py (hunk 1 above)
Comment (andrewlee94) [reply idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py:380]: The state block will contain both true and apparent species. However, the main framework expects there to be a `component_list` which it uses to index the component balances, etc.

If you look at the different tests, we have the ability to set the state block to use either the true or apparent species as the master component list (see line 221 for an example); this case happens to be the true component example. Thus, we want `component_list` to be the same as `true_species_list` for this case. There is another test for the reverse case where `component_list` is the same as `apparent_species_list`.

Comment (TimBartholomew) [reply idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py:380]: Ok makes sense

**Review by TimBartholomew — APPROVED** [review]

**Review by agarciadiego — APPROVED** [review]
LGTM

**Pull request merged**